src/geometry/warp.jl

```diff
@@ -32,18 +32,21 @@ function Homography{T}() where T
     return Homography{T}(SMatrix{3, 3, T, 9}(m))
 end
 
-function (h::Homography{T})(x::SVector{3,K}) where {T,K}
-    y = h.H * x
-    return SVector{2,T}(y[1:2] ./ y[end])
+function (h::Homography{K})(x::SVector{3, L}) where {L, K}
+    hmap = LinearMap(h.H)
+    out = hmap(x)
+    out = out[1:2] / out[3]
+    return SVector{2, K}(out)
 end
 
-function (h::Homography{T})(x::SVector{2,K}) where {T,K}
-    return h(SVector{3,K}(x...,1))
+function (h::Homography{K})(x::AbstractVector{L}) where {L, K}
+    x = SVector(x...)
+    return h(x)
 end
```

Comment:
Actually, I don't think it's type stable here, you might need to restrict it with length-3 vector with some eager check `length(x) == 3`.

```julia
julia> to_svector(x::AbstractVector) = SVector(x...)

julia> @inferred to_svector([1,2])
ERROR: return type SVector{2, Int64} does not match inferred return type Any
Stacktrace:
 [1] error(s::String)
   @ Base ./error.jl:33
 [2] top-level scope
   @ REPL[4]:1

julia> to_svector(x::AbstractVector) = SVector(x...)::SVector{3, eltype(x)}
to_svector (generic function with 1 method)

julia> @inferred to_svector([1,2,3])
3-element SVector{3, Int64} with indices SOneTo(3):
```